tools/automation/building_blocks/device.py

```diff
@@ -0,0 +1,90 @@
+"""
+  Contains classes which are used to store data for a device.
+
+  Ready to use, or extend subclasses as required to fill cases where more 
+  information needs to be stored.
+
+  Call to_dictionary() to retrieve all the data.
+"""
+
+import abc
+
+class Device:
+  
+  class Connection:
+
+    def __init__(self, name, value):
+      self._name = name
+      self._value = value
+
+    def to_dictionary(self):
+      return {self._name: self._value}
+
+  class Translation:
+
+    def __init__(self, name, value):
+      self._name = name
+      self._value = value
+      
+    def to_dictionary(self):
+      return {self._name: self._value}
+
+  class Link:
+
+    def __init__(self, name, value):
+      self._name = name
+      self._value = value
+      
+    def to_dictionary(self):
+      return {self._name: self._value}
+
+  def __init__(self, device_name, device_type, device_id):
+    self._name = device_name
+    self._type = device_type
+    self._id = device_id
+    self._connections = {}
+    self._translation = {}
+    self._links = {}
+  
+  def populate_connections(self, name, value):
+    self._connections.update(self.Connection(name, value).to_dictionary())

```

Comment:
if a connection is just one key value pair, is there a benefit to creating the Connection object first? something like `self._connections[name] = value` seems cleaner and easier to read

same with `populate_translations` and `populate_links`